.github/workflows/ci.yml

```diff
@@ -29,6 +29,7 @@ jobs:
       - name: Install dependencies
         uses: ./.github/actions/build-setup
         with:
+          CONFIG: "{}"
```

Comment:
Does this ever contain any other value?

Reply:
Oh ok. So this here means, only build, don't deploy?

If so, could we have a dedicated input for this?

Reply:
In terms of this here, the action is just the setup portion, before building or deploying. In `check-dependencies`, I can call `build-setup` with an empty config because we don't care about the config part of the setup, just the dependency part.

The reason for calling this reusable `build-setup` action instead of installing dependencies directly in `check-dependencies` is that it's not a one-liner (loading .npmrc, using npm caching, installing). So I find it worth it to avoid redundancy.

In terms of the rest of the workflow, we do have a dedicated input that means only build, don't deploy, it's `workflow_dispatch.inputs.DEPLOY` (when false).

Reply:
If you prefer, I could move the config part out and rename the action `install-dependencies`. It wouldn't match the app anymore, and would have to be updated if we needed to build in another job for any reason, but we wouldn't have to run the config part for `check-dependencies` anymore. Let me know what you think.

Reply:
Your rationale makes sense. I am fine to leave it.

Maybe we can add a comment stating why the config is empty?

```suggestion
          # no actual config required since we are only ensuring that all dependencies can be installed
          CONFIG: "{}"
```

Reply:
Thanks, sounds good. I'll add the comment, good idea.